### PR TITLE
[CI][WIP] integrating error codes utility into CI

### PR DESCRIPTION
**Pull request by DelusionalOptimist**
Signed-off-by: Rudraksh Pareek <rudrakshpareek3601@gmail.com>

**Description**
Integrating the utility into CI to update the error reference at https://docs.meshery.io/reference/error-codes and running checks in PRs.
This PR fixes #

**Notes for Reviewers**
TODOs:
- [x] Support for meshery server.
- [ ] Support for mesheryctl. (Once mesheryctl uses new error codes?).
- [ ] Support for other components (To be done in their repos)
- [ ] Running checks in PRs (keeping out of scope of this PR)

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/error-ref-publisher.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Error Reference Publisher
```
Comment (leecalcote) [inline .github/workflows/error-ref-publisher.yaml:1]:
```suggestion
name: Error Code Reference Publisher
```

File: .github/workflows/error-ref-publisher.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+on:
+  push:
+    branches:
+      - 'master'
+    paths-ignore:
+      - 'docs/**'
+      - 'mesheryctl/**'
```
Comment (leecalcote) [inline .github/workflows/error-ref-publisher.yaml:8]: We will probably want to run this workflow on change to mesheryctl as well.

File: .github/workflows/error-ref-publisher.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          echo '{ "errors_export": '$(cat errorutil_errors_export.json)' }' | jq > docs/_data/errorref/meshery-server_errors_export.json
+
+      - name: Commit changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_user_name: l5io
+          commit_user_email: ci@layer5.io
+          commit_author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+          commit_options: '--signoff'
+          commit_message: "[Docs] update meshery server error reference"
```
Comment (leecalcote) [inline .github/workflows/error-ref-publisher.yaml:34]:
```suggestion
          commit_message: "[Docs] Error Code Reference: update Meshery server error codes"
```

File: .github/workflows/error-ref-publisher.yaml (hunk 2 above)
Comment (DelusionalOptimist) [reply .github/workflows/error-ref-publisher.yaml:8]: I think we should do this once mesheryctl gets updated with new error codes :thinking:

Comment (leecalcote) [reply .github/workflows/error-ref-publisher.yaml:8]: Ok, that's fine. 

For clarity: the error code utility should have idempotent behavior. What harm does enabling it on mesheryctl now cause?

Comment (DelusionalOptimist) [reply .github/workflows/error-ref-publisher.yaml:8]: Hmm yes, no harm. I mixed things up a bit. We can enable this :P
The utility will just throw `ERRO[0000] Unable to execute root command (open component_info.json: no such file or directory)` and exit with status 0. So the workflow will still continue.

**Comment by leecalcote** [comment]
@DelusionalOptimist ready to merge current progress?

**Pull request merged**